src/main/solution_space/solution_graph.py

```diff
@@ -228,3 +231,32 @@ def add_code_info_chain(self, code_info_chain: Optional[List[Tuple[Code, CodeInf
                 prev_vertex.add_child(next_vertex)
                 prev_vertex = next_vertex
         log.info(f'Finish adding code-user chain')
+
+    @staticmethod
+    def update_ages_and_experiences(code_info_list: List[CodeInfo],
+                                    ages: List[int],
+                                    experiences: List[Any]) -> None:
+        for code_info in code_info_list:
+            if not code_info.user.profile.age != DEFAULT_VALUES.AGE.value:
+                ages.append(code_info.user.profile.age)
+            if not code_info.user.profile.experience != DEFAULT_VALUES.EXPERIENCE:
+                experiences.append(code_info.user.profile.experience)
+
+    def calculate_median_of_profile_info(self) -> Tuple[int, Any]:
```

Comment:
I'm not sure that we need a median for a profile in this pr, because we need to have a small and simple implementation of path finder. Maybe it would be better to move them into separate branch and leave for better times?